resources/config_auth_azuread.rb

```diff
@@ -0,0 +1,54 @@
+# Cookbook:: grafana
+# Resource:: config_auth_azuread
+#
+# Copyright:: 2018, Sous Chefs
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Configures the installed grafana instance
+
+property  :instance_name,                                  String,         name_property: true
+property  :azuread_name,                                   String,         default: 'AzureAD'
+property  :azuread_enabled,                                [true, false],  default: false
```

Comment:
because the resource name is azuread, i'd recommend removing that from the prefix of these properties. Just make this `enabled`, `allow_sign_up`, and so on